### DISCUSSION TITLE
Trying to understand how this device has been detected

### DISCUSSION THREAD

**Discussion by badnetmask** [andrew-codechimp/HA-Battery-Notes#1264] [General]
I have an Aeotec MultiSensor 7 that is presented to Home Assistant via the SmartThings integration. On the "Devices" screen, this particular device does show as `<unknown>`, but for some reason HA-Battery-Notes was able to identify the device, and even displays the right battery.

The same SmartThings integration also presents an Aeotec MultiSensor 6, which also shows up as `<unknown>`, but HA-Battery-Notes does not detect this device.

I tried poking around the flat files in Home Assistant and I could not figure how how one device is detected but not the other. Any help here?

I have other sensors that also show `<unknown>`, which are not detected either, so maybe if I figure this one out I can figure out the others as well.

Thank you.

**Comment by andrew-codechimp** [comment]
Battery Notes uses the device manufacturer and model information for the library to be able to detect things.

I don't have any SmartThings devices myself but there are some in the [library](https://github.com/andrew-codechimp/HA-Battery-Notes/blob/main/library.md) with SmartThings as a manufacturer and a model like Multipurpose sensor (2018 model) (IM6001-MPP01)

If you look in your local files at .storage/core.device_registry you will see a section like below for every device you have, if you can identify the device then you will see its manufacturer and model and be able to [submit a new device](https://github.com/andrew-codechimp/HA-Battery-Notes/issues/new?template=new_device_request.yml&title=%5BDevice%5D%3A+) with those details, I don't know why the SmartThings integration chooses not to show these against the device, most integrations do.

A hint to make tracking them down easier, whilst viewing the actual device page the URL will end with the device id like this
https://my-ha/config/devices/device/05d31c2f6f9460ebf21335f67e14ba1f

If all else fails you can set them up manually with the Add Device button within the battery notes configuration screen.

```
     {
        "area_id": null,
        "config_entries": [
          "7f3f2a97efe6cd808b6e7cbd67faa439",
          "fc7d6818c97cdac2e62d61557f309ddd"
        ],
        "disabled_by": null,
        "entry_type": null,
        "hw_version": null,
        "id": "05d31c2f6f9460ebf21335f67e14ba1f",
        "identifiers": [],
        "manufacturer": "Espressif",
        "model": "m5stack-atom",
        "name_by_user": null,
        "name": "Esphm5Matrix01",
        "sw_version": "2023.12.9 (Feb 11 2024, 14:45:15)",
        "via_device_id": null
      }
```

**Comment by badnetmask** [comment]
First, I realized my original post was badly formatted and it was missing the `<unknown>` tags. This has been corrected.

I looked into the library, and those devices listed as "SmartThing" are not the same thing I am referring to. Those are actual SmartThings devices, while the one I am trying to figure out is an Aeotec device connected to a SmartThings hub, connected to Home Assistant via webhook. Different things.

I had look at that device information page in the past, but it doesn't give me the manufacturer and device ID, most likely because of the SmartThings integration problem (like I said, they show up as `<unknown>` on the devices list page).

Looking at the `device_registry` file, they're still unknown, so I'm still baffled. Both devices have the exact same information in the JSON file as you can see below, but one is detected (Sensor1), the other is not (Sensor2):
```
      {
        "area_id": "area1",
        "config_entries": [
          "REDACTED"
        ],
        "configuration_url": "https://account.smartthings.com",
        "connections": [],
        "disabled_by": null,
        "entry_type": null,
        "hw_version": null,
        "id": "REDACTED",
        "identifiers": [
          [
            "smartthings",
            "REDACTED"
          ]
        ],
        "manufacturer": null,
        "model": null,
        "name_by_user": null,
        "name": "Sensor1",
        "serial_number": null,
        "sw_version": null,
        "via_device_id": null
      },
```
Anything else you think I can look at?

**Reply by andrew-codechimp** [reply]
I am at a loss what to suggest on that one as I don't understand how the first was discovered.
The discovery code walks through the device_registry file, looking at manufacturer & model for each device and tries to match against the library list.  Both of those being null will not find a match.  The discovery code only looks at enabled devices, but then that wouldn't help with the nulls.

Are you sure that sensor1 that is identified is not also added via another integration and this entry is a left over that wasn't deleted?

Within core.config_entries you'll find entries for battery notes that tell you the device_id it's associated with as a way of cross referencing to the device_registry like this, you can find sensor1's entry for domain battery_notes to be certain.

```
      {
        "entry_id": "fc7d6818c97cdac2e62d61557f309ddd",
        "version": 2,
        "domain": "battery_notes",
        "title": "Esphm5Matrix01",
        "data": {
          "device_id": "05d31c2f6f9460ebf21335f67e14ba1f",
          "battery_quantity": 2,
          "battery_type": "Rechargeable",
          "battery_low_threshold": 0
        },
        "options": {},
        "pref_disable_new_entities": false,
        "pref_disable_polling": false,
        "source": "user",
        "unique_id": "bn_05d31c2f6f9460ebf21335f67e14ba1f",
        "disabled_by": null
      }
```

**Reply by badnetmask** [reply]
I see. It shows "source: user", which I guess means either me or something else added it manually?
If it was not automatically detected, I didn't do it, and I can't reproduce, then I guess this isn't a bug on your code, and I'll just add the other device manually.
Sorry for the noise.

**Reply by andrew-codechimp** [reply]
The "source: user" is a HA internal thing, I don't record how a device was added myself so the mystery remains.
Sorry I couldn't help on this one, but it was worth the discussion for me to review things so never consider this noise.
